### PR TITLE
feat: Fossfolio V2

### PR DESCRIPTION
**Pull request by DarkPhoenix2704**
- Eslint & Prettier to Biome
- Knex Migration
- Backend structure refactor
- Root Module -> Merged all modules to a single one
- Simplified Folder structure 
- remove direct usage of process.env -> configModule
- Global Error Filter
- Global Error
- Class Validator -> ZodSchema migration (WIP)

### CODE REVIEW

File: apps/api/src/services/events.service.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+				title: 'In Progress',
+			},
+			{
+				fk_event_id: event.id,
+				title: 'Done',
+			},
+		]);
+
+		this.eventEmitter.emit('event.created', event, user.id);
+	}
```
Comment (sreehari2003) [inline apps/api/src/services/events.service.ts:82]: can we convert this to event emitter ?

File: apps/api/src/db/migrations/16.event-ticket.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+export async function down(knex: Knex): Promise<void> {
+  return knex.schema.table(SystemTable.EventTicket, () => {
+    knex.schema
+      .hasColumn(SystemTable.EventTicket, "fk_user_id")
+      .then((exist) => {
+        knex.schema.table(SystemTable.EventTicket, (t) =>
+          t.dropColumn("fk_user_id")
+        );
+      });
+  });
```
Comment (DarkPhoenix2704) [inline apps/api/src/db/migrations/16.event-ticket.ts]: Why adding fk_user_id here? 

These are the multiple ticket types for an event. No need for an fk_user_id

File: apps/api/src/controllers/form.controller.ts [hunk 3]
```diff
@@ -56,10 +56,12 @@ export class FormController {
...
       });
       if (!event || event?.price > 0) {
+        // temporary we dont support paid event
+        // @reenphygeorge
         throw new NotFoundException();
       }
-      await this.form.addUserFormSubmission(data, eventId, user.uid);
-      return await this.events.registerEvent(event.slug, user.uid);
+      await this.form.addUserFormSubmission(data, eventId, user.id);
+      return await this.events.registerEvent(event.slug, user.id);
```
Comment (DarkPhoenix2704) [inline apps/api/src/controllers/form.controller.ts]: In future, we could support allowing registration without logging in.

File: apps/api/src/services/organization.service.ts [hunk 4]
```diff
@@ -13,22 +19,37 @@ export class OrganizationService {
...
+        fk_organization_id: newOrg.id,
+        fk_user_id: uid,
+        role: Role.ADMIN,
+      });
+
+      const getOrgWithMemberInfo = BaseContext.knex(SystemTable.Org)
+        .select("*")
+        .where({
+          name,
+        })
```
Comment (DarkPhoenix2704) [inline apps/api/src/services/organization.service.ts]: name ?

File: apps/api/src/services/organization.service.ts [hunk 5]
```diff
@@ -13,22 +19,37 @@ export class OrganizationService {
...
+        .leftJoin(
+          SystemTable.Org,
+          `${SystemTable.Org}.id`,
+          `${SystemTable.OrgMember}.fk_organization_id`
+        );
```
Comment (DarkPhoenix2704) [inline apps/api/src/services/organization.service.ts]: Move this to orgModel

File: apps/api/src/services/events.service.ts (hunk 1 above)
Comment (DarkPhoenix2704) [reply apps/api/src/services/events.service.ts:84]: let's keep as this now. Will be more predictable this way.

File: apps/api/src/services/auth/strategy/jwt.strategy.ts [hunk 6]
```diff
@@ -35,6 +35,8 @@ export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
 	}) {
 		const user = await UserModel.findById(payload.sub);
 
+		console.log(user)
```
Comment (sreehari2003) [inline apps/api/src/services/auth/strategy/jwt.strategy.ts:38]: remove log

**Comment by sreehari2003** [comment]
frontend format caused this much change , should set up a pre commit hook

File: apps/web/package.json [hunk 7]
```diff
@@ -3,7 +3,7 @@
   "version": "1.0.0",
   "description": "Event management software",
   "scripts": {
-    "dev": "next dev --turbo",
```
Comment (DarkPhoenix2704) [inline apps/web/package.json]: Turbo is production ready for development. 
I don't see any reason to remove them

Comment (DarkPhoenix2704) [reply apps/web/package.json]: @sreehari2003 revert this in next pr

Comment (SreehariJayaraj) [reply apps/web/package.json]: ya frontend was not working for me so i was trying to fix that , sorry about that

**Pull request merged**